webrender/src/clip_scroll_tree.rs

```diff
@@ -338,27 +336,34 @@ impl ClipScrollTree {
                 NodeType::ReferenceFrame(ref info) => TransformUpdateState {
                     parent_reference_frame_transform: node.world_viewport_transform,
                     parent_combined_viewport_rect: node.combined_local_viewport_rect,
-                    parent_scroll_offset: LayerVector2D::zero(),
                     parent_accumulated_scroll_offset: LayerVector2D::zero(),
                     nearest_scrolling_ancestor_viewport: state
                         .nearest_scrolling_ancestor_viewport
                         .translate(&info.origin_in_parent_reference_frame),
                     ..*state
                 },
-                NodeType::Clip(..) | NodeType::StickyFrame(..) => TransformUpdateState {
+                NodeType::Clip(..) => TransformUpdateState {
                     parent_combined_viewport_rect: node.combined_local_viewport_rect,
-                    parent_scroll_offset: LayerVector2D::zero(),
                     ..*state
                 },
                 NodeType::ScrollFrame(ref scrolling) => TransformUpdateState {
-                    parent_combined_viewport_rect: node.combined_local_viewport_rect,
-                    parent_scroll_offset: scrolling.offset,
+                    parent_combined_viewport_rect:
+                        node.combined_local_viewport_rect.translate(&-scrolling.offset),
```

Comment:
wouldn't this change the semantics of scroll frames? I thought the PR only targeting sticky frames logic

Reply:
Previously, `parent_scroll_offset` and  `parent_combined_viewport_rect`  were combined to calculate `scrolled_parent_combined_clip` at https://github.com/servo/webrender/pull/1845/files#diff-6e285308c5268349a08cb330b8f08781L259. Since this no longer happens, we can remove `parent_scroll_offset`. I need to pass along the combined rectangle, because sticky frames need the scroll offset passed along (in `accumulated_scroll_offset`), but `parent_combined_viewport_rect` already had the offset applied as part of the `combined_local_viewport_rect` calculation.

All this means that the semantics of scroll frames don't change, but the value that is passed in `parent_combined_viewport_rect` does. `update_transform` now interprets this values as the parent combined viewport rect in the coordinate system of the children.